pkg/tfbridge/walk.go

```diff
@@ -92,8 +92,10 @@ func propertyPathToSchemaPathInner(
 	}
 
 	// Detect single-nested blocks (object types).
-	if res, isRes := schema.Elem().(shim.Resource); schema.Type() == shim.TypeMap && isRes {
-		return propertyPathToSchemaPath(basePath, propertyPath, res.Schema(), schemaInfo.Fields)
+	//
+	// This is the case where (schema & schema.Elem) ~ {x: T}.
+	if res, isRes := util.CastToTypeObject(schema); isRes {
+		return propertyPathToSchemaPath(basePath, propertyPath, res, schemaInfo.Fields)
```

Comment:
Is this case also covered in the tests? Seems to me like we used to return in more cases here - what are some examples of situation where we used to return here but no longer do that? Should we add some regression tests for these cases?

Reply:
This change is a refactor. It has no behavioral impact. The comment documents existing behavior.